src/main/java/net/dv8tion/jda/api/entities/Widget.java

```diff
@@ -0,0 +1,806 @@
+     * @return possibly-null VoiceChannel with the given ID. 
+     */
+    @Nullable
+    public VoiceChannel getVoiceChannelById(String id)
+    {
+        checkAvailable();
+
+        return channels.get(MiscUtil.parseSnowflake(id));
+    }
+
+    /**
+     * Gets a voice channel with the given ID, or {@code null} if the voice channel is not found
+     *
+     * @param  id
+     *         the ID of the voice channel
+     *
+     * @throws IllegalStateException
+     *         If the widget is not {@link #isAvailable() available}
+     *
+     * @return possibly-null VoiceChannel with the given ID.
+     */
+    @Nullable
+    public VoiceChannel getVoiceChannelById(long id)
+    {
+        checkAvailable();
+
+        return channels.get(id);
+    }
+    
+    /**
+     * Gets a list of online members in the guild
+     *
+     * @throws IllegalStateException
+     *         If the widget is not {@link #isAvailable() available}
+     *
+     * @return the list of members
+     */
+    @Nonnull
+    public List<Member> getMembers()
+    {
+        checkAvailable();
+
+        return Collections.unmodifiableList(new ArrayList<>(members.valueCollection()));
+    }
+    
+    /**
+     * Gets a member with the given ID, or null if the member is not found
+     * 
+     * @param  id
+     *         the ID of the member
+     *
+     * @throws NumberFormatException
+     *         If the provided {@code id} cannot be parsed by {@link Long#parseLong(String)}
+     * @throws IllegalStateException
+     *         If the widget is not {@link #isAvailable() available}
+     *
+     * @return possibly-null Member with the given ID. 
+     */
+    @Nullable
+    public Member getMemberById(String id)
+    {
+        checkAvailable();
+
+        return members.get(MiscUtil.parseSnowflake(id));
+    }
+
+    /**
+     * Gets a member with the given ID, or {@code null} if the member is not found
+     *
+     * @param  id
+     *         the ID of the member
+     *
+     * @throws IllegalStateException
+     *         If the widget is not {@link #isAvailable() available}
+     *
+     * @return possibly-null Member with the given ID.
+     */
+    @Nullable
+    public Member getMemberById(long id)
+    {
+        checkAvailable();
+
+        return members.get(id);
+    }
+
+    @Override
+    public int hashCode() {
+        return Long.hashCode(id);
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (!(obj instanceof Widget))
+            return false;
+        Widget oWidget = (Widget) obj;
+        return this == oWidget || this.id == oWidget.getIdLong();
+    }
+    
+    @Override
+    public String toString()
+    {
+        return "W:" + (isAvailable() ? getName() : "") + '(' + id + ')';
+    }
+
+    private void checkAvailable()
+    {
+        if (!isAvailable)
+            throw new IllegalStateException("The widget for this Guild is unavailable!");
+    }
+
+    public static class Member implements IMentionable
```

Comment:
nested classes could be lifted out as their own classes

Reply:
The nested classes should not be lifted out as their own.
They are attributes of Widget, just like `Invite.Channel` and `Invite.Guild`